sbin/build.sh

```diff
@@ -371,14 +371,36 @@ executeTemplatedFile() {
 
 }
 
+getGradleHome() {
+  local gradleJavaHome=""
+
+  if [ ${JAVA_HOME+x} ]; then
+    gradleJavaHome=${JAVA_HOME}
+  fi
+  if [ ${JDK11_BOOT_DIR+x} ]; then
+    gradleJavaHome=${JDK11_BOOT_DIR}
+  fi
+
+  echo $gradleJavaHome
+}
+
 buildSharedLibs() {
     cd "${LIB_DIR}"
-    GRADLE_USER_HOME=./gradle-cache ./gradlew clean uberjar
+    local gradleJavaHome=$(getGradleHome)
+    echo "Running gradle with $gradleJavaHome"
+
+    JAVA_HOME="$gradleJavaHome" GRADLE_USER_HOME=./gradle-cache ./gradlew clean uberjar
+
+    # Test parser
+    "$gradleJavaHome"/bin/java -version 2>&1 | "$gradleJavaHome"/bin/java -cp "${LIB_DIR}/target/libs/adopt-shared-lib.jar" ParseVersion -s -f semver 1
```

Comment:
Could be clearer as to what's this is testing in the comment, e.g. What's the expected output